sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -166,12 +199,16 @@ public Mono<Void> sendDisposition(String deliveryTag, DeliveryState desiredState
      * @param delivery    the delivery object updated from the broker's transfer frame ack.
      */
     public void onDispositionAck(UUID deliveryTag, Delivery delivery) {
+        // Note: It's by design that this method doesn't check for the 'isTerminated' flag. This ack route needs to
+        // stay open for potential concurrent termination-route awaiting for in-progress dispositions completion/timeout.
+        // termination-route == 'terminateAndAwaitForDispositionsInProgressToComplete'
+
         final DeliveryState remoteState = delivery.getRemoteState();
 
         logger.atVerbose()
             .addKeyValue(DELIVERY_TAG_KEY, deliveryTag)
             .addKeyValue(DELIVERY_STATE_KEY, remoteState)
-            .log("Received update disposition delivery.");
+            .log("onDispositionAck");
 
         final Outcome remoteOutcome;
         if (remoteState instanceof Outcome) {
```

Comment:
Couldn't comment on the exact line (line 225), but I see `.addKeyValue("delivery", delivery)` used in a few places, consider adding a logging key?

Reply:
Yes, will define a key in ClientConstants for "delivery"